galsim/config/value.py

```diff
@@ -219,17 +219,25 @@ def GetCurrentValue(key, config, value_type=None, base=None, return_safe=False):
 
         if chain: 
             # If there are more keys, just set d to the next in the chain.
-            d = d[k]
+            try:
+                d = d[k]
+            except TypeError:
+                raise ValueError("Invalid key in GetCurrentValue = %s"%key)
```

Comment:
I'll add KeyError, since that would also be relevant.  But the type error is to catch the case where `d` is already the final object, say a float or a Position2D.  Then doing `d[k]` raises a TypeError.  And the error is not terribly informative about what when wrong.

But if the next key in the chain is not valid, that would raise a KeyError, which would probably also be worth catching here and telling the user what the real problem is.

Done here: 608d776